Upgrade main to Lucene 10

javanna opened the pull request:
This PR upgrades the main branch to Lucene 10. 

It is work in progress, based on a Lucene 10 snapshot, and should not get merged until Lucene 10 is released.

This is a superset of the changes made to upgrade main to Lucene 9.12 (#113333).

cbuescher committed:
Fix APMTracer by determinizing automata

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-b91b4136aff

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-56468ea3bb8

ChrisHegarty committed:
Add awaits fix bugURL

cbuescher committed:
Removing IndexVersions.V_7_0_0 and IndexVersions.V_7_1_0

cbuescher committed:
Delete IndexVersions.V_7_2_0 - V_7_5_2

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-5f242b3b268

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-40c4e582cf9

ChrisHegarty committed:
Update usage of Automaton Operations.subsetOf in x-pack core

ChrisHegarty committed:
Mute test that uses removed complement syntax

ChrisHegarty committed:
A couple more Automaton subsetOf

ChrisHegarty committed:
Update usage of Automaton sameLanguage in test

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-dc47adbbe73

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-ef5d0f2729a

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-dc47adbbe73

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-371fa57d9c7

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-dc47adbbe73

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-371fa57d9c7

cbuescher committed:
Remove IndexVersions.V_7_6_0

cbuescher committed:
Remove IndexVersions.V_7_7_0 - V_7_11_0

cbuescher committed:
Remove remainign V_7x IndexVersions

cbuescher committed:
Uncomment some tests for better mergability with main

cbuescher committed:
Merge branch 'main' into lucene_snapshot_10

cbuescher committed:
Fix errors in imports

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-ce23e15eb54

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-64f5697f537

cbuescher committed:
Merge branch 'main' into lucene_snapshot_10

cbuescher committed:
Remove usages of random versions before V_8_0_0 in tests

cbuescher committed:
More test fixed related to legacy versions

cbuescher committed:
Mute CompositeRolesStoreTests.testXPackUserCanAccessNonRestrictedIndices

cbuescher committed:
Fix failing knn yaml test

cbuescher committed:
Fix docs test related to knn queries

cbuescher committed:
Null out missing codec in BWCLucene70Codec

This temporarily gets us past missing codec that we attempt to load at node
startup, even if we don't use it.

cbuescher committed:
Partially revert previous commit

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-7964682ddf5

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-7c529ce092d

cbuescher committed:
Fix compilation issues after last Lucene 10 snapshot merge

cbuescher committed:
Merge branch 'main' into lucene_snapshot_10

cbuescher committed:
Fix UOE in ContextIndexSearcher after last Lucene merge

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-74e3c44063a

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-ab262f917d4

cbuescher committed:
Fix compile issues after latest Lucene snapshot update

cbuescher committed:
Merge branch 'main' into lucene_snapshot_10

 Conflicts:
	build-tools-internal/version.properties
	modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/EdgeNGramTokenizerTests.java
	modules/analysis-common/src/test/java/org/elasticsearch/analysis/common/SynonymsAnalysisTests.java
	modules/reindex/src/internalClusterTest/java/org/elasticsearch/migration/MultiFeatureMigrationIT.java
	plugins/analysis-phonetic/src/test/java/org/elasticsearch/plugin/analysis/phonetic/AnalysisPhoneticFactoryTests.java
	qa/rolling-upgrade-legacy/src/test/java/org/elasticsearch/upgrades/RecoveryIT.java
	server/src/internalClusterTest/java/org/elasticsearch/repositories/IndexSnapshotsServiceIT.java
	server/src/test/java/org/elasticsearch/action/admin/cluster/stats/VersionStatsTests.java
	server/src/test/java/org/elasticsearch/cluster/metadata/MetadataCreateIndexServiceTests.java
	server/src/test/java/org/elasticsearch/cluster/metadata/MetadataTests.java
	server/src/test/java/org/elasticsearch/index/IndexVersionTests.java
	server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java
	server/src/test/java/org/elasticsearch/index/mapper/DateFieldMapperTests.java
	server/src/test/java/org/elasticsearch/index/mapper/DynamicTemplatesTests.java
	server/src/test/java/org/elasticsearch/index/mapper/FieldNamesFieldMapperTests.java
	server/src/test/java/org/elasticsearch/index/mapper/ParametrizedMapperTests.java
	server/src/test/java/org/elasticsearch/index/mapper/vectors/BinaryDenseVectorScriptDocValuesTests.java
	server/src/test/java/org/elasticsearch/index/similarity/SimilarityServiceTests.java
	server/src/test/java/org/elasticsearch/indices/analysis/AnalysisModuleTests.java
	server/src/test/java/org/elasticsearch/script/VectorScoreScriptUtilsTests.java
	server/src/test/java/org/elasticsearch/script/field/vectors/DenseVectorTests.java
	test/framework/src/main/java/org/elasticsearch/index/mapper/MetadataMapperTestCase.java
	test/framework/src/main/java/org/elasticsearch/snapshots/AbstractSnapshotIntegTestCase.java
	x-pack/plugin/deprecation/src/test/java/org/elasticsearch/xpack/deprecation/IndexDeprecationChecksTests.java
	x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/mapper/ShapeFieldMapperTests.java
	x-pack/plugin/spatial/src/test/java/org/elasticsearch/xpack/spatial/index/query/LegacyGeoShapeWithDocValuesQueryTests.java

cbuescher committed:
spotless

cbuescher committed:
Fixing more checkstyle and spotless issues after merging main

mayya-sharipova committed:
Remove awaitsFix, as the issue was fixed

javanna committed:
Override the correct search method

search(List<LeafReaderContext>, Weight, Collector) is deprecated and mostly unused.
We need to override the other one that accepts a LeafReaderContextPartition[].

cbuescher committed:
Follow up changes to inter-segment concurrency changes

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-5045d3c67b1

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-6cc4f13ab22

cbuescher committed:
Use RegExp.DEPRECATED_COMPLEMENT where needed

Lucene 10 removed the RegExp.COMPLEMENT flag and the ~ operator but brought it
back as a deprecated option in https://github.com/apache/lucene/pull/13739, so
for the time being we should still be able to use it.

cbuescher committed:
Unmute two tests that now pass

ChrisHegarty committed:
Unmute a couple more tests that now pass

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-7c056ab88c7

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-1b38d5dec85

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-568d1f3fbe7

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-9cd6a24be43

ChrisHegarty committed:
Fix getDiscountOverlaps in LegacyBM25Similarity

ChrisHegarty committed:
Fix AggregatorTestCase with LeafReaderContextPartition

ChrisHegarty committed:
More LeafReaderContextPartition refactoring fixes

Commit:
lucene_snapshot: fix license headers

Commit:
Merge branch 'main' into lucene_snapshot_10

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-a774a998be1

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-3801d859783

Commit:
lucene_snapshot: Fix constructor chaining in LegacyBM25Similarity

Commit:
lucene_snapshot: Fix constructor chaining in LegacyBM25Similarity

Commit:
lucene_snapshot_10: fix license headers

javanna committed:
Move AggregatorTestCase to search(Query, CollectorManager)

search(Query, Collector) is deprecated and no longer performs post collection

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot_10

cbuescher committed:
Don't randomize LuceneTestCase concurrency when using "newSearcher"

LuceneTestCase#newSearcher(reader) now randomizes which concurrency type is
being used. For some parts we don't support INTRA_SEGMENT concurrency, so we
override this widely used utility method so we use INTER_SEGMENT concurrency for
now.

cbuescher committed:
Fix docs according to changes in Lovins token filter, Pathhierarchy Analyzer and Noritoken filter

cbuescher committed:
Fix geo_shape related docs tests

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-f4ebed2404e

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-cd7a74cb4d4

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot_10

cbuescher committed:
Fix expected output of romanian analyzer

cbuescher committed:
Fix QueryTranslatorSpecTests due to changes in regex syntax flags

cbuescher committed:
Fix 370_profile yaml test for yamlRestCompatTest

cbuescher committed:
Fix romanian analyzer restBwc test

cbuescher committed:
Remove v7.17.13 bwc tasks in CI

cbuescher committed:
Determinize automaton produced by IncludeExclude

cbuescher committed:
Fix persian language analyzer doc by adding stemmer

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-71ca6b4bb16

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-b59a357e586

Commit:
lucene_snapshot: fix another instance of IOContext.READONCE

cbuescher committed:
Fix compile errors after L10 snapshot merge

Commit:
Merge branch 'main' into lucene_snapshot_new

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-6d987e1ce1c

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-b467a2bb66d

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

cbuescher committed:
Merge branch 'main' into lucene_snapshot_10

ChrisHegarty committed:
Add a capability and transport version for new regex and range interval query support (#113128)

This commit adds a capability and transport version for new regex and range interval query support.

mayya-sharipova committed:
Multi term intervals: increase max_expansions (#112826)

Currently multi term interval queries (prefix, wildcard, fuzzy, regexp and range) can
expand maximum to 128 terms. To reach parity with span queries that we want to deprecate,
set max expansions to indices.query.bool.max_clause_count which is used in span queries.

Relates to #110491

mayya-sharipova committed:
Multi term intervals: increase max_expansions (#112826)

Currently multi term interval queries (prefix, wildcard, fuzzy, regexp and range) can
expand maximum to 128 terms. To reach parity with span queries that we want to deprecate,
set max expansions to indices.query.bool.max_clause_count which is used in span queries.

Relates to #110491

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-e4ac57746eb

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-a7ce3466d7c

javanna committed:
Adapt QueryAnalyzer to use TermInSetQuery#getBytesRefIterator

javanna committed:
restore ngram tokenizer removed due to a bad merge

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Address norwegian stemmer creation issues

javanna committed:
Merge branch 'main' into lucene_snapshot_10

javanna committed:
extend ESTestCase#newSearcher methods and add javadocs

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Rephrase comment in OldCodecsAvailableTests

This isn't a blocker for the lucene 10 upgrade, it is for the release of stack 9.
It's already annotated @UpdateForV9 so we are good.

javanna committed:
clarify comment in BWCLucene70Codec

Added update for V9 annotation so we don't forget, not a blocker for lucene 10 upgrade

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot_10

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

ChrisHegarty committed:
Fix READONCE IOContext usage

javanna committed:
fix TransportSimulateBulkActionIT compilation

ChrisHegarty committed:
another attempt to fix READONCE IOContext usage

ChrisHegarty committed:
spotless

ChrisHegarty committed:
Update docs/changelog/113018.yaml

ChrisHegarty committed:
Use the RC build

javanna committed:
Address test failures in old-lucene-versions

javanna committed:
Address test failure in BlockPostingsFormat3Tests

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-53d1c2bd2fb

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-11c4f071a7a

ChrisHegarty committed:
Update docs/changelog/113333.yaml

elasticmachine committed:
Merge branch 'main' into lucene_snapshot_9_12

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot_10

ChrisHegarty committed:
remove erroneous changelog

javanna committed:
Address WildcardFieldMapperTests failure

elasticmachine committed:
Merge branch 'main' into lucene_snapshot_9_12

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-11c4f071a7a

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-53d1c2bd2fb

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_9_12

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot_10

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

ChrisHegarty committed:
fix WildcardFieldMapperTests to include

ChrisHegarty committed:
Fix docs build

elasticmachine committed:
Merge branch 'main' into lucene_snapshot_9_12

elasticmachine committed:
Merge branch 'main' into lucene_snapshot_9_12

elasticmachine committed:
Merge branch 'main' into lucene_snapshot_9_12

elasticmachine committed:
Merge branch 'main' into lucene_snapshot_9_12

ChrisHegarty committed:
Merge branch 'lucene_snapshot_9_12' into lucene_snapshot_10

ChrisHegarty committed:
revert

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-53d1c2bd2fb

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-11c4f071a7a

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

javanna committed:
Merge branch 'main' into lucene_snapshot_10

javanna committed:
Restore index versions 7 in lucene_snapshot_10 (#113317)

ChrisHegarty committed:
Add Lucene70DocValuesFormat to old-lucene-versions plugin (#113377)

Add Lucene70DocValuesFormat to old-lucene-versions plugin.

Lucene 10 has remove the old Lucene70DocValuesFormat from its backward_codecs, so we can no longer lookup the doc values format by name (DocValuesFormat.forName("Lucene70")). Instead we make a copy and reference it statically.

The code is for the most part a direct copy, with the following changes:

1. package and import updates
2. Elasticsearch's spotless
3. A couple of changes in Lucene70DocValuesConsumer, where we no longer iterate SortedSetDocValues until NO_MORE_DOCS, but rather use a counted loop with docValueCount. As per what is done elsewhere in lucene 10.
TheLucene70DocValuesFormatTests has been, similar to how we test other formats in this plugin.

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-53d1c2bd2fb

Commit:
[Automated] Update Lucene snapshot to 9.12.0-snapshot-11c4f071a7a

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot_10

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot_10

javanna committed:
Remove leftover TODO

ChrisHegarty committed:
Remove TODO in RegexpFlag

javanna committed:
Add UpdateForV10 annotation

brianseeders committed:
Merge remote-tracking branch 'upstream/lucene_snapshot_10' into lucene_snapshot

javanna committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Revert needless jdk change in legacy file

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-53d1c2bd2fb

javanna committed:
Merge branch 'main' into lucene_snapshot

cbuescher committed:
Fix compile issues after last merge with main

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-ff57fa7b423

cbuescher committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-7b4b0238d70

javanna committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

javanna committed:
update profile tests

We run in a mixed clusters, replacing the profile description isn't enough, we need to skip
the test against previous versions because we get a different output depending on the
version of the node that has the shard.

I isolated the offending assertion to a separate test, so that we don't need to skip the entire profile test.

javanna committed:
add lucene 10 upgrade node feature and fix profile yaml test

javanna committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
restore replaceValueInMatch for profile description tests from 8.x

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-7b4b0238d70

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-0a8604d908c

javanna committed:
Revert "[Automated] Update Lucene snapshot to 10.0.0-snapshot-0a8604d908c"

This reverts commit f93db4961957672dd993bd9cbdc766385fb8860a.

javanna committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-22ac47c07ad

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

javanna committed:
Address compile errors after vector api changes upstream (#113766)

Our lucene_snapshot branch requires updating after https://github.com/apache/lucene/pull/13779

javanna committed:
Update lucene snapshot buildkite config to build from branch_10_0 (#113797)

The Lucene 10 branch has been cut, we want to switch the snapshot creation to the
newly created branch until Lucene 10 has been released. Later, we will switch to branch_10x.

javanna committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Make dutch_kp and lovins no op token filters

original-brownbear committed:
Fix needless use of concurrent collector managers (#113739)

The utility methods for creating the collectors that were removed from
Lucene created non-concurrent collector managers. The default
constructors create concurrent ones though which leads to a performance
regression, especially when sorting.
-> fixed by adjusting to non-concurrent managers again

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-524ea208c87

javanna committed:
Merge branch 'main' into lucene_snapshot

cbuescher committed:
Make "german2" an alias for "german" snowball stemmer (#113614)

With Lucene 10, German2Stemmer, which is used as a parameter for the Snowball
stemmer, has been folded into GermanStemmer. This results mainly in different
treatment Umlauts, i.e.  where formerly "german" would stem "Bücher" -> "Buch"
but "Buecher" -> "Buech" and "german2" would stem both to the same form "Buch",
this is now true for the general "german" stemmer variant.

This change makes the defunct "german2" language stemmer an alias for the
"german" stemmer that now includes the same improved functionality.

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

ChrisHegarty committed:
Fix build IOConext READ -> DEFAULT

ChrisHegarty committed:
Fix bad merge IOContext READ -> DEFAULT

javanna committed:
fix compilation after merge

javanna committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-4461bc1eff4

javanna committed:
Replace Lucene912Codec with Lucene100Codec

See https://github.com/apache/lucene/pull/13815

javanna committed:
Introduce Elasticsearch900Codec

javanna committed:
Add the missing bits for Elasticsearch900Codec

javanna commented on modules/analysis-common/src/yamlRestTest/resources/rest-api-spec/test/analysis-common/20_analyzers.yml:
```diff
@@ -957,7 +957,7 @@
             analyzer: my_analyzer
     - length: { tokens: 1 }
-    - match:  { tokens.0.token: absenţ }
+    - match:  { tokens.0.token: absenț }
```
@cbuescher I assume that these tests don't run in a mixed cluster , because they are in analysis-common, hence not part of the ordinary yaml tests? Otherwise, I think that a cluster feature should have been used in here, because this test would fail in a mixed cluster? This is different from the rest compat stuff that you addressed in the build file by replacing tokens.

cbuescher replied:
I think so, yes. But I haven't checked it, I assume something would fail on the lucene_snapshot branch otherwise

javanna committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

cbuescher committed:
Conditional stemming for 'persian' analyzer (#113482)

The 'persian' analyzer for Lucene 10 comes with PersianStemFilter as the last
token filter by default. In order to maintain compatibility for old indices, we 
use the new analyzer for new indices but configure a legacy analyzer that is
used with existing indices.

cbuescher committed:
Add bwc layer for 'romanian' analyzer (#113906)

The 'romanian' language analyzer has been improved in Lucene 10 in two important
ways. First, the snowball stemmer has been modified to work with s-comma and t-comma characters
but only with their cedilla forms used when Romanian didn't have full Unicode
support (https://github.com/snowballstem/snowball/pull/177). Second, the
analyzer now contains a normalization step to map cedilla forms to forms with comma.

In order to maintain backwards compatibility with existing indices, this change
moves the Lucene 9 stemmer over to the analysis module was a deprecated variant
and creates the analyzer for existing indices with the "old" stemmer and without
the normalization step. New indices automatically run with the improved
behaviour.

javanna committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Fix bad merge

javanna committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
minor changes to changelog for persian analyzer

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-f76fdb293e1

pmpailis committed:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

javanna committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-f76fdb293e1

pmpailis committed:
muting RankDocsQueryBuilderTests testRankDocsQueryEarlyTerminate

cbuescher committed:
Merge branch 'main' into lucene_snapshot

cbuescher committed:
Fix compilation issue

cbuescher committed:
Add changelog about the nori dictionary update (#114124)

javanna committed:
fix typos

pmpailis committed:
Updating RankDocRetrieverBuilderIT for lucene_snapshot branch (#114098)

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-a4c0f741ccc

cbuescher committed:
Add Snowball upgrade changelog (#114146)

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-a4c0f741ccc

ChrisHegarty committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-a4c0f741ccc

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-a4c0f741ccc

pmpailis committed:
muting RankDocsQueryBuilderTests testRankDocsQueryEarlyTerminate (#114198)

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

javanna committed:
Delete full cluster restart test removed in main

This was restored due to a bad merge.

cbuescher committed:
Remove smoke test section in MetadataCreateIndexServiceTests

javanna commented on server/src/main/java/org/elasticsearch/index/IndexVersions.java:
```diff
@@ -118,6 +128,9 @@ private static IndexVersion def(int id, Version luceneVersion) {
     public static final IndexVersion UPGRADE_TO_LUCENE_9_12 = def(8_516_00_0, Version.LUCENE_9_12_0);
     public static final IndexVersion ENABLE_IGNORE_ABOVE_LOGSDB = def(8_517_00_0, Version.LUCENE_9_12_0);
+
+    public static final IndexVersion UPGRADE_TO_LUCENE_10_0_0 = def(9_000_00_0, Version.LUCENE_10_0_0);
```
@mark-vieira can you double check that this all looks correct to you. The parseUnchecked method is to parse older lucene versions until unsupported index versions are removed.

mark-vieira replied:
@rjernst My understanding is that we didn't want to actually bump the major version on index or transport versions until we cut the branch for the final 8.x release. Is that correct?

javanna replied:
I was wondering about the version bump as well, thanks for calling it out explicitly.

thecoop replied:
Yes, lucene 10 should have index version 9000000. Any index version changes that need to be backported to 8.x will need to be handled similar to transport version backports (ie very carefully)

ChrisHegarty replied:
Transport Version - there are no serialization format changes in this PR, so no transport version change is necessary.

We absolutely need to an Index Version bump - because we're changing the index format.

javanna committed:
spotless

javanna committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Address EngineTestCase compile errors

javanna committed:
Address more compile errors around TotalHitCountCollectorManager constructor

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-a4c0f741ccc

javanna committed:
Merge branch 'main' into lucene_snapshot

cbuescher committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Merge branch 'main' into lucene_snapshot

benwtrent committed:
Fix bbq for Lucene 10

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-a4c0f741ccc

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-bc478d85a12

ChrisHegarty committed:
apply MADV_NORMAL advice to enable more aggressive readahead (#114410)

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-eadc07cc6a1

cbuescher committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Remove empty queue conditional from slicing logic (#114513)

With recent changes in Lucene around not forking execution when not necessary, we have
removed the search worker thread pool in #111099. The worker thread pool had unlimited queue,
and the fear was that we couuld have much more queueing on the search thread pool if we use
it to parallelize execution across segments, because every shard would take up to a thread
per slice when executing the query phase.

We have then introduced an additional conditional to stop parallelizing when there is a queue.
That is perhaps a bit extreme, as it's a decision made when creating the searcher, while a
queue may no longer be there once the search is executing. This has caused some benchmarks
regressions, hence this commit removes the additional queue dependent conditional in order
to perform additional benchmarks without it.

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-eadc07cc6a1

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

cbuescher committed:
Merge branch 'main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-eadc07cc6a1

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-eadc07cc6a1

Commit:
Merge remote-tracking branch 'origin/main' into lucene_snapshot

Commit:
[Automated] Update Lucene snapshot to 10.0.0-snapshot-eadc07cc6a1

cbuescher committed:
Merge branch 'main' into lucene_snapshot

javanna committed:
Merge branch 'main' into lucene_snapshot

javanna commented:
Will open a new PR now that Lucene 10 is out, from a new branch that targets the released artifact.
